examples/dmabuf-capture.c

```diff
@@ -490,10 +491,8 @@ static void *vid_encode_thread(void *arg) {
 		}
 
 		while (1) {
-			AVPacket pkt;
-			av_init_packet(&pkt);
-
-			int ret = avcodec_receive_packet(ctx->avctx, &pkt);
+			pkt = av_packet_alloc();
+			int ret = avcodec_receive_packet(ctx->avctx, pkt);
 			if (ret == AVERROR(EAGAIN)) {
 				break;
```

Comment:
We still leak from here as ctx->err is not set (assuming that the write below here doesn't fail for zero-sized packets).

Reply:
Eh, indeed. I've just reverted to free'ing the packet inside each branch.